flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/operators/rank/RetractableTopNFunction.java

```diff
@@ -374,10 +381,10 @@ private void retractRecordWithoutRowNumber(
 			} else if (findsSortKey) {
 				long count = entry.getValue();
 				long tmpRank = curRank + count;
-				if (isInRankEnd(tmpRank)) {
+				if (tmpRank < rankEnd) {
```

Comment:
Could you add some comments on this? IIUC, this is for padding the last entry because we retract a entry from the top-n. 
Besides, could you improve the local field name of `tmpRank`? It's not easy to understand what the rank it is.